retrieval/k8s/k8s_stats.go

```diff
@@ -178,154 +177,211 @@ func shouldSkipPod(previousHour time.Time, resource *corev1.Pod) bool {
+	case *corev1.Service:
+		cast := to.(*corev1.Service)
+		trimMeta(&cast.ObjectMeta)
+		return cast
+	case *corev1.ReplicationController:
+		cast := to.(*corev1.ReplicationController)
+		trimMeta(&cast.ObjectMeta)
+		return cast
+	case *corev1.Namespace:
+		return trimNamespace(to)
+	case *corev1.PersistentVolume:
+		cast := to.(*corev1.PersistentVolume)
+		trimMeta(&cast.ObjectMeta)
 		return cast
 	case *corev1.PersistentVolumeClaim:
 		cast := to.(*corev1.PersistentVolumeClaim)
-		sanitizeMeta(&cast.ObjectMeta, parseMetricsData)
+		trimMeta(&cast.ObjectMeta)
 		return cast
 	case *corev1.Node:
 		cast := to.(*corev1.Node)
-		sanitizeMeta(&cast.ObjectMeta, parseMetricsData)
+		trimMeta(&cast.ObjectMeta)
 		return cast
 	}
 	return to
 }
 
-func sanitizeMeta(objectMeta *metav1.ObjectMeta, parseMetricsData bool) {
+func sanitizeMeta(objectMeta *metav1.ObjectMeta) {
+	objectMeta.ManagedFields = nil
+	delete(objectMeta.Annotations, KubernetesLastAppliedConfig)
+	objectMeta.Finalizers = nil
+}
+
+func trimMeta(objectMeta *metav1.ObjectMeta) {
 	objectMeta.ManagedFields = nil
 	delete(objectMeta.Annotations, KubernetesLastAppliedConfig)
-	if parseMetricsData {
-		objectMeta.Finalizers = nil
-	}
 }
 
-func sanitizePod(to interface{}, parseMetricsData bool) interface{} {
+func sanitizePod(to interface{}) interface{} {
 	cast := to.(*corev1.Pod)
 
 	// stripping env var and related data from the object
 	(*cast).ObjectMeta.ManagedFields = nil
 	delete((*cast).ObjectMeta.Annotations, KubernetesLastAppliedConfig)
 
 	for j, container := range (*cast).Spec.Containers {
-		(*cast).Spec.Containers[j] = sanitizeContainer(container, parseMetricsData)
+		(*cast).Spec.Containers[j] = sanitizeContainer(container)
 	}
 	for j, container := range (*cast).Spec.InitContainers {
-		(*cast).Spec.InitContainers[j] = sanitizeContainer(container, parseMetricsData)
+		(*cast).Spec.InitContainers[j] = sanitizeContainer(container)
 	}
 	return cast
 }
 
-func sanitizeContainer(container corev1.Container, parseMetricsData bool) corev1.Container {
-	container.Env = nil
-	if parseMetricsData {
-		container.Command = nil
-		container.Args = nil
-		container.ImagePullPolicy = ""
-		container.LivenessProbe = nil
-		container.StartupProbe = nil
-		container.ReadinessProbe = nil
-		container.TerminationMessagePath = ""
-		container.TerminationMessagePolicy = ""
-		container.SecurityContext = nil
+func trimPod(to interface{}) interface{} {
+	cast := to.(*corev1.Pod)
+	// removing env var and related data from the object
+	(*cast).ObjectMeta.ManagedFields = nil
+	delete((*cast).ObjectMeta.Annotations, KubernetesLastAppliedConfig)
+
+	for j, container := range (*cast).Spec.Containers {
+		(*cast).Spec.Containers[j] = trimContainer(container)
+	}
+	for j, container := range (*cast).Spec.InitContainers {
+		(*cast).Spec.InitContainers[j] = trimContainer(container)
 	}
+	return cast
+}
+
+func sanitizeContainer(container corev1.Container) corev1.Container {
+	container.Env = nil
```

Comment:
can remove anything in the `sanitize` funcs that's already in the `trim` funcs

Reply:
Updated